File: Javascript/MonteCarlo/approxpi.js


function fire(){
    var a = Math.random();
    var b = Math.random();
    return {hit: a*a + b*b < 1}
}

var shots = [];
var max = 1000000;
for (var i = 0; i <= max; i++) {
    shots.push(fire());
}
var hits = shots.filter(function(shot){return shot.hit;}).length;
console.log(hits/max * 4);

File: Nim/MonteCarlo/approxpi.nim
import math, sequtils
randomize()

type
    Shot = enum
        miss, hit

proc fire() : Shot =
    let a = random(1.0)
    let b = random(1.0)
    if (a*a+b*b<1): result = hit
    else: result = miss

let max = 1000000
var shots = newSeq[Shot](max)
shots.mapIt(fire())
let hits = shots.filterIt(it == hit).len
echo (hits/max*4)